api/db/transactions.go

```diff
@@ -28,24 +27,25 @@ func (tm *transactionManager) DB() *gorm.DB {
 
 // Transaction embed a callback in a db transaction
 // if the callback returns an error, the transaction is rollbacked
-func (tm *transactionManager) Transaction(callback func(db *gorm.DB) error) error {
-	tm.log.Debugf("opening transaction")
+func (tm *transactionManager) Transaction(callback func(db *gorm.DB) error, log logrus.FieldLogger) error {
+	log.Debug("opening transaction")
 	tx := tm.db.Begin()
+	tx.SetLogger(getLogger(log))
 	if err := tx.Error; err != nil {
 		return err
 	}
 	if err := callback(tx); err != nil {
 		if err := tx.Rollback().Error; err != nil {
-			tm.log.WithError(err).Warnf("transaction has not been rollbacked")
+			log.WithError(err).Warn("transaction has not been rollbacked")
 			return err
 		}
-		tm.log.Warnf("transaction has been rollbacked")
+		log.Warnf("transaction has been rollbacked")
 		return err
 	}
-	tm.log.Debugf("committing transaction")
+	log.Debug("committing transaction")
 	err := tx.Commit().Error
-	if err != nil {
-		tm.log.Debugf("transaction has ben committed")
+	if err == nil {
```

Comment:
We were emitting erroneous logs for transactions ?

Reply:
nop, but the debug log on transaction commit success was happening only on commit failure 😁